### ISSUE TITLE
scv.utils.clean_obs_names() not working properly after updating to latest github version

### ISSUE THREAD

**Issue by JesseRop** [theislab/scvelo#843]
scv.utils.clean_obs_names() not properly extracting barcodes from obs names with particular patterns

I'm sorry but I'm unable to generate an anndata matrix to make a reprex but the code works for the obs_names in the first set and not for the second set

```pytb
Index(['possorted_genome_bam_H66NQ:AAAGAACAGTCAAGCGx',
       'possorted_genome_bam_H66NQ:AAAGAACAGACATATGx',
       'possorted_genome_bam_H66NQ:AAAGAACAGCATTTGCx'],
      dtype='object', name='CellID')
```

```pytb
Index(['possorted_genome_bam_7YCS2:AAACCCAGTCGGCTACx',
       'possorted_genome_bam_7YCS2:AAACGCTTCAATCGGTx',
       'possorted_genome_bam_7YCS2:AAAGAACCATAAGCAAx'],
      dtype='object', name='CellID')
```

<details> <summary> Versions </summary>

```pytb
scvelo==0.2.5.dev68+g9bfd9fb  scanpy==1.8.2  anndata==0.7.8  loompy==3.0.6  numpy==1.21.2  scipy==1.7.1  matplotlib==3.4.3  sklearn==0.24.2  pandas==1.3.2  
```

**Comment by WeilerP** [comment]
@JesseRop, what exactly is the bug here? Could you please share a code snippet, traceback and expected behavior?

**Comment by GabrielBaldissera** [comment]
I am experiencing the same thing. I was using an older version and I got the developer one.  The function used to generate a dataframe of barcodes as it does for the second sample. For some reason, it's not doing it for the first one although the input is still the same I used last year with an older version.


```

scv.utils.clean_obs_names(sample_JG12_WT)
sample_JG12_WT.obs
```

> Out[35]: 
>                                         sample_batch
> KM        possorted_genome_bam_1OU:AAACCCATCGGCATTAx
> KM-1      possorted_genome_bam_1OU:AAAGAACGTCAGGCAAx
> KM-2      possorted_genome_bam_1OU:AAAGGATGTGAATGATx
> KM-3      possorted_genome_bam_1OU:AAAGAACGTGTCTCCTx
> KM-4      possorted_genome_bam_1OU:AAAGGGCGTCCAGCACx


```
scv.utils.clean_obs_names(sample_JG13_MT)
sample_JG13_MT.obs
```

> Out[37]: 
> Empty DataFrame
> Columns: []
> Index: [AAATGGAAGTTATGGA, AAACCCAGTCAAAGTA, AAACGCTCAACCGTGC, AAACCCACATCGAAGG, AACAAAGAGAAACACT, , ...]
> [9788 rows x 0 columns]

How can I correct this so that the first sample becomes like the second? I would like to use the newest version because of the plotting improvements.

Thanks

**Comment by GabrielBaldissera** [comment]
> I am experiencing the same thing. I was using an older version and I got the developer one. The function used to generate a dataframe of barcodes as it does for the second sample. For some reason, it's not doing it for the first one although the input is still the same I used last year with an older version.
> 
> ```
> 
> scv.utils.clean_obs_names(sample_JG12_WT)
> sample_JG12_WT.obs
> ```
> 
> > Out[35]:
> > sample_batch
> > KM        possorted_genome_bam_1OU:AAACCCATCGGCATTAx
> > KM-1      possorted_genome_bam_1OU:AAAGAACGTCAGGCAAx
> > KM-2      possorted_genome_bam_1OU:AAAGGATGTGAATGATx
> > KM-3      possorted_genome_bam_1OU:AAAGAACGTGTCTCCTx
> > KM-4      possorted_genome_bam_1OU:AAAGGGCGTCCAGCACx
> 
> ```
> scv.utils.clean_obs_names(sample_JG13_MT)
> sample_JG13_MT.obs
> ```
> 
> > Out[37]:
> > Empty DataFrame
> > Columns: []
> > Index: [AAATGGAAGTTATGGA, AAACCCAGTCAAAGTA, AAACGCTCAACCGTGC, AAACCCACATCGAAGG, AACAAAGAGAAACACT, , ...]
> > [9788 rows x 0 columns]
> 
> How can I correct this so that the first sample becomes like the second? I would like to use the newest version because of the plotting improvements.
> 
> Thanks

I solved this by installing Version 0.2.4 . Thanks

**Comment by WeilerP** [comment]
Fixed by @921.
